engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/CallActivityInterruptionTest.java

```diff
@@ -102,13 +102,14 @@ public void shouldTriggerBoundaryEventIfChildIsCompleting() {
     // then
     assertThat(
             RecordingExporter.records()
+                .onlyEvents()
                 .betweenProcessInstance(processInstanceKey)
                 .processInstanceRecords())
         .extracting(r -> tuple(r.getValue().getBpmnElementType(), r.getIntent()))
         .containsSubsequence(
-            tuple(BpmnElementType.CALL_ACTIVITY, ProcessInstanceIntent.TERMINATE_ELEMENT),
             tuple(BpmnElementType.CALL_ACTIVITY, ProcessInstanceIntent.ELEMENT_TERMINATING),
-            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATING),
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATED),
```

Comment:
💭 This failing test case made me wonder why it failed at all. It failed because the child PROCESS was ELEMENT_COMPLETED, but now it is ELEMENT_TERMINATED. Why is that the case?

Well, the COMPLETE_ELEMENT command written on line 95 sets in motion the completion of the child process in several steps:
- first completing the task
- then activating and completing the end event
- and finally completing the process. 

💡 Simultaneously, the Timer TRIGGER command sets in motion an interruption, also in several steps. However, the completion of the child process takes more steps now, because the End Event is no longer activated and completed in a single step but instead in an additional step. This is why the Process does not make it in time to be completed and instead is terminated.

❌ The test is called `shouldTriggerBoundaryEventIfChildIsCompleting` which makes me wonder whether the test is simply no longer valid in its current form. The child is not completing when the boundary event triggers. 

💭 However, setting up this scenario in a test is really difficult. I have managed through some very hacky code that I'm not sure we want to have in our test code base.

🔧 I want to propose a different solution:

Let's not write `COMPLETE_ELEMENT` for End Event, but let's keep the `onComplete()` method implemented. The `onActivate()` can be implemented like:

```java
public void onActivate(final ExecutableEndEvent element, final BpmnElementContext activating) {
  final var activated = stateTransitionBehavior.transitionToActivated(activating);
  final var completing = stateTransitionBehavior.transitionToCompleting(activated);
  onComplete(element, completing);
}
```

This also requires a small change to the `OutputMappingIncidentTest` lines 137-142:
```java
final var failureElementInstance =
    RecordingExporter.processInstanceRecords()
        .withElementId(elementId)
        .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETING)
        .withProcessInstanceKey(processInstanceKey)
        .getFirst();
```

💭 This way, we can avoid having to adjust any test cases. Note that we should reconsider the `CallActivityInterruptionTest` at some point because it's hard to set up this test scenario. But let's keep that out of scope for now.